Watchman.Engine/Generation/Generic/CloudFormationAlarmCreator.cs

```diff
@@ -73,20 +73,28 @@ public async Task SaveChanges(bool dryRun)
 
                 var stackName = StackName(alertingGroup);
 
-                try
-                {
-                    await GenerateAndDeployStack(
-                        alarms,
-                        alertingGroup.Targets,
-                        alertingGroup.Name,
-                        stackName,
-                        dryRun);
-                }
-                catch (Exception e)
+
+                for (int i = 0; i < alertingGroup.NumberOfCloudFormationStacks; i++)
```

Comment:
minor
wonder if it might be a bit cleaner to take the alarms and GroupBy the results of Bucket()

then iterate the groups to deploy each stack. you would then have access to the group index and could remove `i` and `iClosure`

Reply:
good call

Reply:
Tried to clean this up but it failed situations where we have no alarms (thank god for end to end tests), so just left it as it was.